### DISCUSSION TITLE
SPDIF and I2S simultaneously?;

### DISCUSSION THREAD

**Discussion by JayShoe** [pschatzmann/arduino-audio-tools#909] [Q&A] [answered]
Hello, I noticed that the audio tools mentioned that the SPDif output is essentially the I2S data pin. I'm curious how many i2S output data pins are available on the esp32 and is it possible to use both at the same time?

I want to send I2S to a DAC and also SPDif to a separate amplifier at the same time. In fact I'll need one i2S input, an i2S output and SPDIF output. For bonus I would also need SPDIF input.

**Comment by pschatzmann** [comment, marked as answer]
The ESP32 has 2 I2S ports: each port is duplex so you can read and write at the same time...

**Reply by JayShoe** [reply]
Thank you, Phil.

Can you please tell me if the Arduino audio tools has any provisions for TDM?

 Specifically configuring i2S as TDM and i2S2 has SPDIF?

I'm also curious if there is 8x8 ADAT over SPDIF/Toslink?

**Comment by pschatzmann** [comment]
Not sure if the regular ESP32 supports TDM. 
In any case this would be only available in the new API which is not available in Arduino (yet)
So, no I am not supporting this...

You can try to work with IDF and uncomment and correct the DRAFT TDM implementation in I2SESP32-New.h.

**Reply by JayShoe** [reply]
Ok sounds awesome. I am working on a product that marries the ESP32 with the teensy audio library. It would be interesting to somehow combine the efforts your putting into this with the work going on over there. Regardless, your effort is appreciated. I'll keep you posted on my progress.
